src/storage/documentService/documentService.js

```diff
@@ -48,80 +32,85 @@ const getDocument = async (id, { cleanup } = { cleanup: false }) => {
   }
   return document;
 };
-// We define this function separately to getDocument as we don't want a future dev to accidentally return a decryption key by using the wrong function
+
 const getDecryptionKey = async id => {
   const params = {
-    TableName: config.dynamodb.storageTableName,
-    Key: {
-      id
-    }
+    Bucket: config.bucketName,
+    Key: id
   };
   const document = await get(params);
+  if (!document.key) throw new Error("The conditional request failed");
   return document;
 };
 
-const validateTtl = ttl => {
-  if (typeof ttl !== "number" || ttl < 0)
-    throw new Error("TTL must be a positive number of seconds");
-  if (ttl > MAX_TTL) throw new Error("TTL exceeds maximum of 30 days");
-  return true;
-};
-
-const uploadDocument = async (
+const uploadDocumentAtId = async (
   document,
   documentId,
-  ttl = DEFAULT_TTL,
   network = config.network
 ) => {
+  const placeHolderObj = await getDecryptionKey(documentId);
+  if (!(placeHolderObj.key && placeHolderObj.awaitingUpload)) {
+    // we get here when a file exists at location but is not a placeholder awaiting upload
+    throw new Error(`No placeholder file`);
+  }
+
```

Comment:
I think we can put this operation after verifying the document. As reading and writing cost on S3.

Reply:
verifying is very slow though i prefer to keep the execution time as low as possible, don't think cost is a big issue for now

Reply:
$0.09/gb i think its ok